### PR TITLE
fix: correct partition-id to field-id in UnboundPartitionField

### PR DESCRIPTION
**Pull request by FANNG1**
fixes: #575

### CODE REVIEW

File: crates/iceberg/src/spec/table_metadata.rs [hunk 1]
```diff
@@ -1293,7 +1293,7 @@ mod tests {
                 name: "x".to_string(),
                 transform: Transform::Identity,
                 source_id: 1,
-                partition_id: Some(1000),
+                field_id: Some(1000),
```
Comment (Xuanwo) [inline crates/iceberg/src/spec/table_metadata.rs:1296]: I'm a bit surprised that the test against `testdata/table_metadata/TableMetadataV2Valid.json` passed.

Comment (FANNG1) [reply crates/iceberg/src/spec/table_metadata.rs:1296]: sure

Comment (FANNG1) [reply crates/iceberg/src/spec/table_metadata.rs:1296]: When build PartitionSpec, `UnboundPartitionField` is transformed to `PartitionField`,  in which `partition_id` is transformed `field_id` 
```rust
            bound_fields.push(PartitionField {
                source_id: field.source_id,
                field_id: partition_id,
                name: field.name,
                transform: field.transform,
            })
```
https://github.com/apache/iceberg-rust/blob/main/crates/iceberg/src/spec/partition.rs#L415-L420

Comment (FANNG1) [reply crates/iceberg/src/spec/table_metadata.rs:1296]: By the way, I think `add_partition_field` is more general to use, seems `add_unbound_field` should only be used in test to specify partition field id.

Comment (Xuanwo) [reply crates/iceberg/src/spec/table_metadata.rs:1296]: Thanks, would you like to submit a new issue for disucsion. I believe this PR is good to go.

**Review by Xuanwo — APPROVED** [review]
Thanks for the fix!

File: crates/iceberg/src/spec/table_metadata.rs (hunk 1 above)
Comment (FANNG1) [reply crates/iceberg/src/spec/table_metadata.rs:1296]: ok, thanks

**Pull request merged**